doc/SAI-Proposal-IPv6_Segment_Routing-1.md

```diff
@@ -0,0 +1,207 @@
+
+For the second mechanism, the route / next-hop group / next-hop tables are used to identify the SR DIP and manipulate via endpoint function or source/transit SID lists as objects in the next-hop table.  Afterwards the flow will head to the egress pipeline for futher lookup on the modified packet.  Thus, the only structural changes needed is in the next-hop table to handle the transit, source, and endpoint actions.
+
+The below figures shows the additional logic in the Ingress ACL and Route lookup in the behavioral pipeline to support this.
+
+![SAI v6SR bm](figures/sai_v6SR_bm.png "Figure 1: Behavioral Model Addition. ")
+__Figure 1: Behavioral Model Addition.__
+
+![SAI v6SR bm1](figures/sai_v6SR_bm1.png "Figure 2: Source Behavior ")
+__Figure 2: Source Behavior.__
+
+![SAI v6SR bm2](figures/sai_v6SR_bm2.png "Figure 3: Transit Behavior ")
+__Figure 3: Transit Behavior.__
+
+![SAI v6SR bm3](figures/sai_v6SR_bm3.png "Figure 4: Endpoint Behavior ")
+__Figure 4: Endpoint Behavior.__
+
+
+## API Modification
+
+### Next-Hop Table Modifications
+#### Action Parameters
+
+Set SID List Object ID for transit / source behavior to be taken:
+    SAI_NEXT_HOP_ATTR_SEGMENTROUTE_SIDLIST_ID
+
+Endpoint Actions to be taken:
+    SAI_NEXT_HOP_ATTR_SEGMENTROUTE_ENDPOINT_TYPE
+    SAI_NEXT_HOP_ATTR_SEGMENTROUTE_ENDPOINT_POP_TYPE
+
+> Note: Not all endpoint actions included
+
+### SID List Object APIs
+#### Vendor Support Advertisement
+
+Included is also a way for vendors to advertise device support include the number of segments and TLV types that can be originated
+
+    SAI_SWITCH_ATTR_SEGMENTROUTE_MAX_SEGMENTS
+    SAI_SWITCH_ATTR_SEGMENTROUTE_MAX_SIDLISTS
+    SAI_SWITCH_ATTR_SEGMENTROUTE_TLV_TYPE
+
+> Note: NSH Carrier and Padding TLVs were not included in this first draft
+
+#### Action Parameters
+
+Transit or Source Action to be taken with policy
+
+    SAI_SEGMENTROUTE_SIDLIST_ATTR_TYPE
+
+SID List Action to be taken:
+
+    SAI_SEGMENTROUTE_SIDLIST_TYPE_INSERT
+    SAI_SEGMENTROUTE_SIDLIST_TYPE_ENCAPS
+
+List of DIP segments or TLVs to be added
+
+    SAI_SEGMENTROUTE_SIDLIST_ATTR_SEGMENT_LIST
+    SAI_SEGMENTROUTE_SIDLIST_ATTR_TLV
+
+### Segment Route Counter Support
+
+> Note: No counters included in first draft
+
+### APIs
+
+To start with, the basic create/remove entry and set/get attributes APIs are included
+
+    create_segmentroute_sidlist;
+    remove_segmentroute_sidlist;
+    set_segmentroute_sidlist_attribute;
+    get_segmentroute_sidlist_attribute;
+    create_segmentroute_sidlists;
+    remove_segmentroute_sidlists;
```

Comment:
If that's the naming convention we are following, then I'm fine with it. I would have picked a different one, but the important thing here is that we keep consistency across modules, so i'm good.